qui/updater/summary_page.py

```diff
@@ -0,0 +1,331 @@
+# coding=utf-8
+#
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# Copyright (C) 2023  Piotr Bartman <prbartman@invisiblethingslab.com>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+import asyncio
+import threading
+from typing import Optional, Any
+
+from qubes_config.widgets.gtk_utils import load_icon
+from qubesadmin import exc
+from qubesadmin.events.utils import wait_for_domain_shutdown
+
+from qubes_config.widgets.utils import get_boolean_feature
+from qui.updater.utils import disable_checkboxes, pass_through_event_window, \
+    HeaderCheckbox, QubeClass, QubeLabel, QubeName, Theme, \
+    RowWrapper, ListWrapper, on_head_checkbox_toggled
+
+from locale import gettext as l
+
+
+class SummaryPage:
+    """
+    Last content page of updater.
+
+    Show the summary of vm updates and appms that should be restarted.
+    """
+
+    def __init__(
+            self,
+            builder,
+            theme,
+            next_button,
+            cancel_button,
+            back_by_row_selection
+    ):
+        self.builder = builder
+        self.theme = theme
+        self.next_button = next_button
+        self.cancel_button = cancel_button
+        self.restart_thread = None
+        self.disable_checkboxes = False
+
+        self.updated_tmpls: Optional[list] = None
+
+        self.restart_list = self.builder.get_object("restart_list")
+        self.list_store: Optional[ListWrapper] = None
+
+        self.stack = self.builder.get_object("main_stack")
```

Comment:
as above: please annotate those with types, mypy is unable to parse .glade to guess them